pkg/controller/instance/instance_controller.go

```diff
@@ -126,17 +127,34 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 					planName = "upgrade"
 				}
 			} else if !reflect.DeepEqual(old.Spec, new.Spec) {
-				_, ok = fv.Spec.Plans["update"]
-				if !ok {
-					_, ok = fv.Spec.Plans["deploy"]
-					if !ok {
-						log.Println("InstanceController: Could not find any plan to use for update")
-					} else {
-						planName = "deploy"
+				for k, v := range new.Spec.Parameters {
```

Comment:
Side note: would it make sense to have tests for such more complex behavior?

Reply:
Absolutely.  I'm not sure if its better to have
1) A few examples in config/samples that are used by us and/or a test suite
2) have these hard coded by unit tests
3) both

Reply:
3 ;-). In this case I would probably start with 2...